gradle/docker.gradle.kts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2020 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+import com.bmuschko.gradle.docker.tasks.image.DockerRemoveImage
+import com.bmuschko.gradle.docker.tasks.image.DockerSaveImage
+
+val dockerBaseImageRef = "$buildDir/.docker/buildBaseImage-imageId.txt"
+val dockerBaseAlpineImageRef = "$buildDir/.docker/buildBaseAlpineImage-imageId.txt"
+
+val dockerBaseImageId = "${project.property("docker.domain")}/knotx:${project.version}"
+val dockerBaseAlpineImageId = "${project.property("docker.domain")}/knotx-alpine:${project.version}"
+
+val dockerfileBaseImagePath = "$projectDir/src/main/docker/base/Dockerfile"
+val dockerfileBaseAlpineImagePath = "$projectDir/src/main/docker/base-alpine/Dockerfile"
+
+tasks.register<Copy>("copyDockerfileBase") {
+    group = "docker"
+    from(dockerfileBaseImagePath)
+    into("$buildDir/out/")
+    mustRunAfter("downloadBaseDistribution")
+}
+
+tasks.register<Copy>("copyDockerfileBaseAlpine") {
+    group = "docker"
+    from(dockerfileBaseAlpineImagePath)
+    into("$buildDir/out/")
+    mustRunAfter("downloadBaseDistribution")
+}
+
+tasks.register<DockerRemoveImage>("removeBaseImage") {
+    group = "docker"
+    val spec = object : Spec<Task> {
+        override fun isSatisfiedBy(task: Task): Boolean {
+            return File(dockerBaseImageRef).exists()
+        }
+    }
+    onlyIf(spec)
+
+    targetImageId(if (File(dockerBaseImageRef).exists()) File(dockerBaseImageRef).readText() else "")
+    onError {
+        if (!this.message!!.contains("No such image"))
+            throw this
+    }
+}
+
+tasks.register<DockerRemoveImage>("removeBaseAlpineImage") {
+    group = "docker"
```

Comment:
Looks like a good candidate for `DockerRemoveImage.remove` helper (both tasks have similar logic).

Reply:
@Skejven Check my solution for this tip and let me know if that was the case.